File: cplusplus/src/verify_signature.cpp
#include "autograph/verify_signature.h"

#include "sodium.h"

bool verify_signature(const unsigned char *public_key,
                      const unsigned char *message,
                      const unsigned long long message_size,
                      const unsigned char *signature) {
  int result =
      crypto_sign_verify_detached(signature, message, message_size, public_key);
  return result == 0;
}

File: typescript/src/verify-signature.ts
import { verify } from 'stedy'
import { importPublicSignKey } from './import-key'

const verifySignature = async (
  message: BufferSource,
  publicKey: BufferSource,
  signature: BufferSource
) => verify(message, await importPublicSignKey(publicKey), signature)

export default verifySignature
